vllm/entrypoints/openai/serving_engine.py

```diff
@@ -617,17 +617,16 @@ def _validate_input(
             if token_num >= self.max_model_len:
                 raise ValueError(
                     f"This model's maximum context length is "
-                    f"{self.max_model_len} tokens. However, you requested "
-                    f"{token_num} tokens in the messages, "
-                    f"Please reduce the length of the messages.")
```

Comment:
i think we should move this check outside the `if max_tokens is None:` . we need to check this `token_num >= self.max_model_len` first, regardless of max_tokens.